WordPress/Classes/ViewRelated/Aztec/New Group/AztecVerificationPromptHelper.swift

```diff
@@ -0,0 +1,107 @@
+        guard action == .publish else {
+            return false
+        }
+
+        return !wpComAccount.emailVerified.boolValue
+    }
+
+    /// - parameter presentingViewController: UIViewController that the prompt should be presented from.
+    /// - parameter then: Completion callback to be called after the user dismisses the prompt.
+    /// **Note**: The callback fires only when the user tapped "OK" or we silently verified the account in background. It isn't fired when user attempts to resend the verification email.
+    func displayVerificationPrompt(from presentingViewController: UIViewController,
+                                   then: AztecVerificationPromptCompletion?) {
+
+        let fancyAlert = FancyAlertViewController.verificationPromptController { [weak self] in
+            then?(self?.wpComAccount.emailVerified.boolValue ?? false)
+        }
+
+        fancyAlert.modalPresentationStyle = .custom
+        fancyAlert.transitioningDelegate = self
+        presentingViewController.present(fancyAlert, animated: true)
+
+        updateVerificationStatus()
+        // Silently kick off the request to make sure the user still actually needs to be verified.
+        // If in the meantime user has been verified, we'll dismiss the prompt,
+        // call the completion block and let caller handle the new situation.
+
+        displayedAlert = fancyAlert
+        completionBlock = then
+    }
+
+    func updateVerificationStatus() {
+        accountService.updateUserDetails(for: wpComAccount,
+                                         success: { [weak self] in
+
+                                            // Let's make sure the alert is still on the screen and
+                                            // the verification status has changed, before we call the callback.
+                                            guard let displayedAlert = self?.displayedAlert,
+                                                  let updatedAccount = self?.accountService.defaultWordPressComAccount(),
+                                                  updatedAccount.emailVerified.boolValue else {
+                                                        return
+                                            }
+
+                                            displayedAlert.dismiss(animated: true, completion: nil)
+                                            self?.completionBlock?(updatedAccount.emailVerified.boolValue)
+            }, failure: nil)
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+}
+
+// MARK: - UIViewControllerTransitioningDelegate
+//
+extension AztecVerificationPromptHelper: UIViewControllerTransitioningDelegate {
+    func presentationController(forPresented presented: UIViewController, presenting: UIViewController?, source: UIViewController) -> UIPresentationController? {
+        if presented is FancyAlertViewController {
```

Comment:
Probably a bit nitpicky of me since I don't think we have anything formal in our style guide about it, but we tend to `guard` and return early for a failure scenario (in this case `nil`).

Reply:
To be perfectly honest, I stole that piece of code  from `NUXAbstractViewController`. Updated in both places :)

Reply:
Doh!!  Shame on us!